idx/memory/memory.go

```diff
@@ -45,9 +45,11 @@ type Tree struct {
 }
 
 type Node struct {
-	Path     string
-	Children []string
-	Leaf     bool
+	Path        string
+	Children    []string
+	Defs        []string
+	Leaf        bool
+	HasChildren bool
 }
```

Comment:
sounds fine by me.  and those two concepts are independent from each other right?